genesyscloud/telephony_providers_edges_edge_group/genesyscloud_telephony_providers_edges_edge_group_proxy.go

```diff
@@ -0,0 +1,172 @@
+	return p.getEdgeGroupByNameAttr(ctx, p, edgeGroupName, managed)
+}
+
+func getEdgeGroupByNameFn(ctx context.Context, p *edgeGroupProxy, edgeGroupName string, managed bool) (*[]platformclientv2.Edgegroup, error) {
+	const pageSize = 100
+	var pageNum = 1
+	var allEdgeGroups []platformclientv2.Edgegroup
+
+	edgeGroups, _, err := p.edgesApi.GetTelephonyProvidersEdgesEdgegroups(pageSize, pageNum, edgeGroupName, "", managed)
+	if err != nil {
+		return nil, err
+	}
+	if edgeGroups.Entities != nil && len(*edgeGroups.Entities) > 0 {
+
+		allEdgeGroups = append(allEdgeGroups, (*edgeGroups.Entities)[0])
+
+	}
+
+	for pageNum := 2; pageNum <= *edgeGroups.PageCount; pageNum++ {
+		edgeGroups, _, err := p.edgesApi.GetTelephonyProvidersEdgesEdgegroups(pageSize, pageNum, edgeGroupName, "", managed)
+		if err != nil {
+			return nil, err
+		}
+		if edgeGroups.Entities == nil || len(*edgeGroups.Entities) == 0 {
+			break
+		}
+		allEdgeGroups = append(allEdgeGroups, (*edgeGroups.Entities)[0])
+
+	}
+
+	return &allEdgeGroups, nil
+}
+
+func getEdgeGroupByIdFn(ctx context.Context, p *edgeGroupProxy, edgeGroupId string) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error) {
+	edgeGroup, resp, err := p.edgesApi.GetTelephonyProvidersEdgesEdgegroup(edgeGroupId, nil)
+	if err != nil {
+		return nil, resp, err
+	}
+	return edgeGroup, resp, nil
+}
+
+func deleteEdgeGroupFn(ctx context.Context, p *edgeGroupProxy, edgeGroupId string) (*platformclientv2.APIResponse, error) {
+	resp, err := p.edgesApi.DeleteTelephonyProvidersEdgesEdgegroup(edgeGroupId)
+	return resp, err
+}
+
+func updateEdgeGroupFn(ctx context.Context, p *edgeGroupProxy, edgeGroupId string, body platformclientv2.Edgegroup) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error) {
+	edgeGroup, resp, err := p.edgesApi.PutTelephonyProvidersEdgesEdgegroup(edgeGroupId, body)
+	if err != nil {
+		return nil, resp, err
+	}
+	return edgeGroup, resp, nil
+}
+
+func createEdgeGroupFn(ctx context.Context, p *edgeGroupProxy, body platformclientv2.Edgegroup) (*platformclientv2.Edgegroup, *platformclientv2.APIResponse, error) {
+	edgeGroup, resp, err := p.edgesApi.PostTelephonyProvidersEdgesEdgegroups(body)
+	if err != nil {
+		return nil, resp, err
+	}
+	return edgeGroup, resp, nil
+}
+
+func getAllEdgeGroupsFn(ctx context.Context, p *edgeGroupProxy) (*[]platformclientv2.Edgegroup, error) {
+	const pageSize = 100
+	var allEdgeGroups []platformclientv2.Edgegroup
+	var pageNum = 1
+
+	edgeGroups, _, err := p.edgesApi.GetTelephonyProvidersEdgesEdgegroups(pageSize, pageNum, "", "", false)
+	if err != nil {
+		return nil, err
+	}
+	if edgeGroups.Entities != nil && len(*edgeGroups.Entities) > 0 {
+		for _, edgeGroup := range *edgeGroups.Entities {
+			if edgeGroup.State != nil && *edgeGroup.State != "deleted" {
+				allEdgeGroups = append(allEdgeGroups, edgeGroup)
+			}
+		}
+	}
+
+	if *edgeGroups.PageCount < 2 {
```

Comment:
You don't need this if statement, if there's no page then the loop won't run

Reply:
addressed